weldx/core.py

```diff
@@ -480,9 +484,14 @@ def _init_expression(self, data):
 
     def _interp_time_discrete(self, time: Time) -> xr.DataArray:
         """Interpolate the time series if its data is composed of discrete values."""
+        data = self._data
+        if self.time is None and time.is_absolute:
+            # type: ignore[union-attr]
+            data = data.weldx.reset_reference_time(time.reference_time)
```

Comment:
This is something we should probably discuss. It fixes the following problem:

We replaced

~~~ python
        return ut.xr_interp_like(
            self._data,
            {"time": time.as_data_array()},
            method=self.interpolation,
            assume_sorted=False,
            broadcast_missing=False,
        )
~~~

with

~~~ python
return ut.xr_interp_like(
            data,
            time.as_data_array(),
            method=self.interpolation,
            assume_sorted=False,
            broadcast_missing=False,
        )
~~~

to get absolute times working correctly when using `xr_interp_like`. However, this causes the test case `(ts_const, time_mul + pd.Timestamp("2020"), [1, 1, 1, 1, 1, 1, 1, 1], "m")` of `test_interp_time` to fail (currently line 450 in `test_core.py`). The test just checks if the value of a constant `TimeSeries` that gets interpolated with an array of times is broadcasted correctly. Since the value is constant, reference times shouldn't matter, but after the change, they suddenly did. So I caught this special case and just added a reference time to a copy of the data prior to interpolation. The question is if we want to keep it that way, should it be handled differently or is that something that `xr_interp_like` should handle internally